src/FileSystem/DefaultFileSystem.cpp

```diff
@@ -72,6 +77,26 @@ std::filesystem::path DefaultFileSystem::FindPath(const std::filesystem::path& p
 	throw std::runtime_error("File " + path.string() + " not found");
 }
 
+bool DefaultFileSystem::IsPathValid(const std::filesystem::path& path)
```

Comment:
The way this is implemented requires the game path to be already set (`Exists(path)`). This is a problem because this needs to be able to be called before setting the game path.

The whole point of this function is to verify is a path is valid so we can then call the set to game path.